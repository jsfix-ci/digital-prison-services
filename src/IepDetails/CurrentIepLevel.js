import React from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import GridRow from '@govuk-react/grid-row'
import GridCol from '@govuk-react/grid-col'
import CurrentIepLevelArea from './IepDetails.styles'

const CurrentIepLevel = ({ level, days, nextReviewDate }) => (
  <CurrentIepLevelArea className="current-iep">
    <GridRow>
      <GridCol setWidth="one-quarter">
        <strong className="label">Current IEP level</strong>
        <p>{level}</p>
      </GridCol>
      <GridCol setWidth="one-quarter">
        <strong className="label">Time since review</strong>
        <p>{days}</p>
      </GridCol>
      <GridCol setWidth="one-quarter">
        <strong className="label">Date of next review</strong>
        <p>{nextReviewDate}</p>
      </GridCol>
    </GridRow>
  </CurrentIepLevelArea>
)

CurrentIepLevel.propTypes = {
  level: PropTypes.string,
  days: PropTypes.string,
  nextReviewDate: PropTypes.string,
}

CurrentIepLevel.defaultProps = {
  level: '',
  days: '',
  nextReviewDate: '',
}

const mapStateToProps = state => ({
  level: state.iepHistory.currentIepLevel,
  days: state.iepHistory.daysOnIepLevel,
  nextReviewDate: state.iepHistory.nextReviewDate,
})

export default connect(mapStateToProps)(CurrentIepLevel)
