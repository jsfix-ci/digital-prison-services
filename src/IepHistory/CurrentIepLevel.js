import React from 'react'
import PropTypes from 'prop-types'
import GridRow from '@govuk-react/grid-row'
import GridCol from '@govuk-react/grid-col'
import { CurrentIepLevelArea } from './IepHistory.styles'

const CurrentIepLevel = ({ level, days, nextReviewDate }) => (
  <CurrentIepLevelArea>
    <GridRow>
      <GridCol setWidth="one-quarter">
        <strong className="label">Current IEP level</strong>
        <div>{level}</div>
      </GridCol>
      <GridCol setWidth="one-quarter">
        <strong className="label">Days since review</strong>
        <div>{days}</div>
      </GridCol>
      <GridCol setWidth="one-quarter">
        <strong className="label">Date of next review</strong>
        <div>{nextReviewDate}</div>
      </GridCol>
    </GridRow>
  </CurrentIepLevelArea>
)

CurrentIepLevel.propTypes = {
  level: PropTypes.string,
  days: PropTypes.number,
  nextReviewDate: PropTypes.string,
}

CurrentIepLevel.defaultProps = {
  level: '',
  days: 0,
  nextReviewDate: '',
}

export default CurrentIepLevel
