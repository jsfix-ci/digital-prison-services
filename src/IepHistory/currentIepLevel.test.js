import React from 'react'
import { shallow } from 'enzyme'
import CurrentIepLevel from './CurrentIepLevel'

describe('Current IEP Level', () => {
  it('should render the page correctly', () => {
    const wrapper = shallow(<CurrentIepLevel level="Basic" days={20} nextReviewDate="20/09/2019" />)

    expect(wrapper).toMatchSnapshot()
  })
})
