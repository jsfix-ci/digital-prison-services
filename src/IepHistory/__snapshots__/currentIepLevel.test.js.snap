// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Current IEP Level should render the page correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CurrentIepLevel
    days={20}
    level="Basic"
    nextReviewDate="20/09/2019"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <GridRow>
        <GridCol
          columnFull={false}
          columnOneHalf={false}
          columnOneQuarter={false}
          columnOneThird={false}
          columnThreeQuarters={false}
          columnTwoThirds={false}
          setDesktopWidth={undefined}
          setWidth="one-quarter"
        >
          <strong
            className="label"
          >
            Current IEP level
          </strong>
          <div>
            Basic
          </div>
        </GridCol>
        <GridCol
          columnFull={false}
          columnOneHalf={false}
          columnOneQuarter={false}
          columnOneThird={false}
          columnThreeQuarters={false}
          columnTwoThirds={false}
          setDesktopWidth={undefined}
          setWidth="one-quarter"
        >
          <strong
            className="label"
          >
            Days since review
          </strong>
          <div>
            20
          </div>
        </GridCol>
        <GridCol
          columnFull={false}
          columnOneHalf={false}
          columnOneQuarter={false}
          columnOneThird={false}
          columnThreeQuarters={false}
          columnTwoThirds={false}
          setDesktopWidth={undefined}
          setWidth="one-quarter"
        >
          <strong
            className="label"
          >
            Date of next review
          </strong>
          <div>
            20/09/2019
          </div>
        </GridCol>
      </GridRow>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <GridCol
            columnFull={false}
            columnOneHalf={false}
            columnOneQuarter={false}
            columnOneThird={false}
            columnThreeQuarters={false}
            columnTwoThirds={false}
            setDesktopWidth={undefined}
            setWidth="one-quarter"
          >
            <strong
              className="label"
            >
              Current IEP level
            </strong>
            <div>
              Basic
            </div>
          </GridCol>,
          <GridCol
            columnFull={false}
            columnOneHalf={false}
            columnOneQuarter={false}
            columnOneThird={false}
            columnThreeQuarters={false}
            columnTwoThirds={false}
            setDesktopWidth={undefined}
            setWidth="one-quarter"
          >
            <strong
              className="label"
            >
              Days since review
            </strong>
            <div>
              20
            </div>
          </GridCol>,
          <GridCol
            columnFull={false}
            columnOneHalf={false}
            columnOneQuarter={false}
            columnOneThird={false}
            columnThreeQuarters={false}
            columnTwoThirds={false}
            setDesktopWidth={undefined}
            setWidth="one-quarter"
          >
            <strong
              className="label"
            >
              Date of next review
            </strong>
            <div>
              20/09/2019
            </div>
          </GridCol>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <strong
                className="label"
              >
                Current IEP level
              </strong>,
              <div>
                Basic
              </div>,
            ],
            "columnFull": false,
            "columnOneHalf": false,
            "columnOneQuarter": false,
            "columnOneThird": false,
            "columnThreeQuarters": false,
            "columnTwoThirds": false,
            "setDesktopWidth": undefined,
            "setWidth": "one-quarter",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Current IEP level",
                "className": "label",
              },
              "ref": null,
              "rendered": "Current IEP level",
              "type": "strong",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Basic",
              },
              "ref": null,
              "rendered": "Basic",
              "type": "div",
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <strong
                className="label"
              >
                Days since review
              </strong>,
              <div>
                20
              </div>,
            ],
            "columnFull": false,
            "columnOneHalf": false,
            "columnOneQuarter": false,
            "columnOneThird": false,
            "columnThreeQuarters": false,
            "columnTwoThirds": false,
            "setDesktopWidth": undefined,
            "setWidth": "one-quarter",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Days since review",
                "className": "label",
              },
              "ref": null,
              "rendered": "Days since review",
              "type": "strong",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": 20,
              },
              "ref": null,
              "rendered": 20,
              "type": "div",
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <strong
                className="label"
              >
                Date of next review
              </strong>,
              <div>
                20/09/2019
              </div>,
            ],
            "columnFull": false,
            "columnOneHalf": false,
            "columnOneQuarter": false,
            "columnOneThird": false,
            "columnThreeQuarters": false,
            "columnTwoThirds": false,
            "setDesktopWidth": undefined,
            "setWidth": "one-quarter",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Date of next review",
                "className": "label",
              },
              "ref": null,
              "rendered": "Date of next review",
              "type": "strong",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "20/09/2019",
              },
              "ref": null,
              "rendered": "20/09/2019",
              "type": "div",
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bxivhb",
        "isStatic": true,
        "rules": Array [
          "
  margin-bottom: ",
          "20px",
          ";
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bxivhb",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <GridRow>
          <GridCol
            columnFull={false}
            columnOneHalf={false}
            columnOneQuarter={false}
            columnOneThird={false}
            columnThreeQuarters={false}
            columnTwoThirds={false}
            setDesktopWidth={undefined}
            setWidth="one-quarter"
          >
            <strong
              className="label"
            >
              Current IEP level
            </strong>
            <div>
              Basic
            </div>
          </GridCol>
          <GridCol
            columnFull={false}
            columnOneHalf={false}
            columnOneQuarter={false}
            columnOneThird={false}
            columnThreeQuarters={false}
            columnTwoThirds={false}
            setDesktopWidth={undefined}
            setWidth="one-quarter"
          >
            <strong
              className="label"
            >
              Days since review
            </strong>
            <div>
              20
            </div>
          </GridCol>
          <GridCol
            columnFull={false}
            columnOneHalf={false}
            columnOneQuarter={false}
            columnOneThird={false}
            columnThreeQuarters={false}
            columnTwoThirds={false}
            setDesktopWidth={undefined}
            setWidth="one-quarter"
          >
            <strong
              className="label"
            >
              Date of next review
            </strong>
            <div>
              20/09/2019
            </div>
          </GridCol>
        </GridRow>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <GridCol
              columnFull={false}
              columnOneHalf={false}
              columnOneQuarter={false}
              columnOneThird={false}
              columnThreeQuarters={false}
              columnTwoThirds={false}
              setDesktopWidth={undefined}
              setWidth="one-quarter"
            >
              <strong
                className="label"
              >
                Current IEP level
              </strong>
              <div>
                Basic
              </div>
            </GridCol>,
            <GridCol
              columnFull={false}
              columnOneHalf={false}
              columnOneQuarter={false}
              columnOneThird={false}
              columnThreeQuarters={false}
              columnTwoThirds={false}
              setDesktopWidth={undefined}
              setWidth="one-quarter"
            >
              <strong
                className="label"
              >
                Days since review
              </strong>
              <div>
                20
              </div>
            </GridCol>,
            <GridCol
              columnFull={false}
              columnOneHalf={false}
              columnOneQuarter={false}
              columnOneThird={false}
              columnThreeQuarters={false}
              columnTwoThirds={false}
              setDesktopWidth={undefined}
              setWidth="one-quarter"
            >
              <strong
                className="label"
              >
                Date of next review
              </strong>
              <div>
                20/09/2019
              </div>
            </GridCol>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                <strong
                  className="label"
                >
                  Current IEP level
                </strong>,
                <div>
                  Basic
                </div>,
              ],
              "columnFull": false,
              "columnOneHalf": false,
              "columnOneQuarter": false,
              "columnOneThird": false,
              "columnThreeQuarters": false,
              "columnTwoThirds": false,
              "setDesktopWidth": undefined,
              "setWidth": "one-quarter",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Current IEP level",
                  "className": "label",
                },
                "ref": null,
                "rendered": "Current IEP level",
                "type": "strong",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Basic",
                },
                "ref": null,
                "rendered": "Basic",
                "type": "div",
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                <strong
                  className="label"
                >
                  Days since review
                </strong>,
                <div>
                  20
                </div>,
              ],
              "columnFull": false,
              "columnOneHalf": false,
              "columnOneQuarter": false,
              "columnOneThird": false,
              "columnThreeQuarters": false,
              "columnTwoThirds": false,
              "setDesktopWidth": undefined,
              "setWidth": "one-quarter",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Days since review",
                  "className": "label",
                },
                "ref": null,
                "rendered": "Days since review",
                "type": "strong",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": 20,
                },
                "ref": null,
                "rendered": 20,
                "type": "div",
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                <strong
                  className="label"
                >
                  Date of next review
                </strong>,
                <div>
                  20/09/2019
                </div>,
              ],
              "columnFull": false,
              "columnOneHalf": false,
              "columnOneQuarter": false,
              "columnOneThird": false,
              "columnThreeQuarters": false,
              "columnTwoThirds": false,
              "setDesktopWidth": undefined,
              "setWidth": "one-quarter",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Date of next review",
                  "className": "label",
                },
                "ref": null,
                "rendered": "Date of next review",
                "type": "strong",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "20/09/2019",
                },
                "ref": null,
                "rendered": "20/09/2019",
                "type": "div",
              },
            ],
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bxivhb",
          "isStatic": true,
          "rules": Array [
            "
  margin-bottom: ",
            "20px",
            ";
",
          ],
        },
        "displayName": "styled.div",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-bxivhb",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
